File: solutions/fsharp/cars-assemble/1/CarsAssemble.fs
module CarsAssemble

let baseProduction: float = 221

let successRate (speed: int): float =
    (100.0 - float (if speed > 4 then (10 + (if speed > 8 then (10 + (if speed = 10 then 3 else 0)) else 0)) else 0)) / 100.0

let productionRatePerHour (speed: int): float =
    (float speed) * (successRate speed) * baseProduction

let workingItemsPerMinute (speed: int): int =
    int ((productionRatePerHour speed) / 60.0)

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{

    public static double BaseProductionRatePerHour() => 221.0;
    public static double SuccessRate(int speed)
    {
        if (speed == 0) return 0.0;
        if (speed < 5) return 1.0;
        if (speed < 9) return 0.9;
        if (speed == 9) return 0.8;
        return 0.77;
    }
    
    public static double ProductionRatePerHour(int speed) => SuccessRate(speed) * speed * BaseProductionRatePerHour();

    public static int WorkingItemsPerMinute(int speed) => (int) (ProductionRatePerHour(speed) / 60);
}
